# based on vehicle_interface.param.template.yaml
lgsvl_interface:
  ros__parameters:
    cycle_time_ms: 10
    # Only one of the three control command topics need be specified
    # high_level_command:
    #   name: "high_level_command"
    # basic_command:
    #   name: "vehicle_command"
    raw_command:
      name: "raw_command"
    state_command:
      name: "state_command"
    odometry:
      name: "odometry"
    state_report:
      name: "state_report"
    lgsvl:
      control_command_topic: "lgsvl/vehicle_control_cmd"
      state_command_topic: "lgsvl/vehicle_state_cmd"
      state_report_topic: "lgsvl/state_report"
      odometry_topic: "lgsvl/gnss/gnss_odom"
      kinematic_state_topic: "vehicle_kinematic_state"
      throttle:
        domain: [0.0, 3.0]
        range: [0.0, 100.0]
      brake:
        domain: [-3.0, 0.0]
        range: [100.0, 0.0]
      steer:
        domain: [-0.331, 0.331]
        range: [-100.0, 100.0]
    state_machine:
      gear_shift_velocity_threshold_mps: 0.5
      acceleration_limits:
        min: -3.0
        max: 3.0
        threshold: 1.0
      front_steer_limits:
        min: -0.331
        max: 0.331
        threshold: 0.3
      time_step_ms: 100
      timeout_acceleration_mps2: 3.0
      state_transition_timeout_ms: 3000
      gear_shift_accel_deadzone_mps2: 0.5
